bound/bound.go

```diff
@@ -4,7 +4,41 @@
 
 package bound
 
+import (
+	"math"
+	"sort"
+)
+
 // Bound represents [Min, Max] bounds.
 type Bound struct {
 	Min, Max float64
 }
+
+// Intersection returns intersected bounds between bounds.
+func Intersection(bounds ...Bound) []Bound {
```

Comment:
I think the idea was to just get the single intersection.

`func Intersection(bounds ...Bound) Bound`

Then there's no need to sort or copy. 

You'll also need to add documentation on what happens when len(bound) == 0, where I would say the zero `Bound` should be returned.

Reply:
If we intend to get the single intersection.
what about the update the signature into `func Intersection(b0, b1Bound) Bound`?
IMHO, variadic input intends multiple bounds so the number of intersected bounds can be more than 1.

WDYT?